internal/provider/provider.go

```diff
@@ -301,30 +301,30 @@ type VRFRequest struct {
 	ProviderConfig *ProviderConfig
 }
 
-// PIMProvider is the interface for the realization of the PIM objects over different providers.
-type PIMProvider interface {
+type VTEPProvider interface {
 	Provider
 
-	// EnsurePIM call is responsible for PIM realization on the provider.
-	EnsurePIM(context.Context, *EnsurePIMRequest) error
-	// DeletePIM call is responsible for PIM deletion on the provider.
-	DeletePIM(context.Context, *DeletePIMRequest) error
-}
-
-type EnsurePIMRequest struct {
-	PIM            *v1alpha1.PIM
-	Interfaces     []PIMInterface
-	ProviderConfig *ProviderConfig
+	// EnsureVRF call is responsible for VRF realization on the provider.
+	EnsureVTEP(context.Context, *VTEPRequest) error
+	// DeleteVRF call is responsible for VRF deletion on the provider.
+	DeleteVTEP(context.Context, *VTEPRequest) error
+	// GetInterfaceStatus call is responsible for retrieving the current status of the Interface from the provider.
+	GetStatusVTEP(context.Context, *VTEPRequest) (VTEPStatus, error)
```

Comment:
```suggestion
	GetVTEPStatus(context.Context, *VTEPRequest) (VTEPStatus, error)
```
nit: Would follow this naming scheme.